plugins/destination/meilisearch/client/transform.go

```diff
@@ -2,179 +2,141 @@ package client
-		res[i] = el.String()
-	}
-	return res
-}
-
-func (*Client) TransformUUID(v *schema.UUID) any {
-	if v.Status != schema.Present {
-		return nil
+	for i, row := range rows {
+		for k, v := range m {
+			row[k] = v[i]
+		}
 	}
-	return v.String()
-}
 
-func (*Client) ReverseTransformValues(table *schema.Table, values []any) (schema.CQTypes, error) {
-	res := make(schema.CQTypes, len(values))
-
-	for i, v := range values {
-		typ := table.Columns[i].Type
-		// uuid, cidr, inet, macaddr & string arrays are written as []string
-		// The default implementation will write them as coma-separated strings surrounded by "{}".
-		switch typ {
-		case schema.TypeUUIDArray,
-			schema.TypeCIDRArray,
-			schema.TypeInetArray,
-			schema.TypeMacAddrArray,
-			schema.TypeStringArray:
-			vTyped := v.([]any)
-			parts := make([]string, len(vTyped))
-			for idx, el := range vTyped {
-				parts[idx] = el.(string)
+	return rows
+}
+
+func reverseTransform(builder array.Builder, val any) error {
+	if val == nil {
+		builder.AppendNull()
+		return nil
+	}
+
+	switch builder := builder.(type) {
+	case *array.BooleanBuilder:
+		builder.Append(val.(bool))
+	case *array.Int8Builder:
+		builder.Append(int8(val.(float64)))
+	case *array.Int16Builder:
+		builder.Append(int16(val.(float64)))
+	case *array.Int32Builder:
+		builder.Append(int32(val.(float64)))
+	case *array.Int64Builder:
+		builder.Append(int64(val.(float64)))
+	case *array.Uint8Builder:
+		builder.Append(uint8(val.(float64)))
+	case *array.Uint16Builder:
+		builder.Append(uint16(val.(float64)))
+	case *array.Uint32Builder:
+		builder.Append(uint32(val.(float64)))
+	case *array.Uint64Builder:
+		builder.Append(uint64(val.(float64)))
+	case *array.Float32Builder:
+		builder.Append(float32(val.(float64)))
+	case *array.Float64Builder:
+		builder.Append(val.(float64))
+	case *array.StringBuilder:
+		builder.Append(val.(string))
+	case *array.LargeStringBuilder:
+		builder.Append(val.(string))
+	case *array.BinaryBuilder:
+		var data []byte
+		val := val.(string)
+		if val != "null" {
+			err := json.Unmarshal([]byte(strconv.Quote(val)), &data)
+			if err != nil {
+				return err
 			}
-			v = "{" + strings.Join(parts, ",") + "}"
 		}
-		t := schema.NewCqTypeFromValueType(typ)
-		// update can move value type to another value, so don't worry about err
-		// example: str -> bool, and the destination still works fine
-		_ = t.Set(v)
-		res[i] = t
+		builder.Append(data)
+	case *array.TimestampBuilder:
+		ts, err := arrow.TimestampFromString(val.(string), arrow.Microsecond)
+		if err != nil {
+			return err
+		}
+		builder.Append(ts)
+	case *types.InetBuilder:
+		_, ipNet, err := net.ParseCIDR(val.(string))
+		if err != nil {
+			return err
+		}
+		builder.Append(*ipNet)
+	case *types.JSONBuilder:
+		builder.Append(val)
+	case *types.MacBuilder:
+		mac, err := net.ParseMAC(val.(string))
+		if err != nil {
+			return err
+		}
+		builder.Append(mac)
+	case *types.UUIDBuilder:
+		uid, err := uuid.Parse(val.(string))
```

Comment:
Same. can be safely deleted as there is no special type for `uuid` in meilisearch and `default` case will take care of that

Reply:
Addressed in cb52e9b